### ISSUE TITLE
Pass new N3 tests

### ISSUE THREAD

**Issue by RubenVerborgh** [rdfjs/N3.js#192]

**Comment by william-vw** [comment]
Do you happen to have a grammar specification (e.g., EBNF)? ..

**Comment by RubenVerborgh** [comment]
Not really, I started from https://www.w3.org/TR/turtle/#sec-grammar and combined https://www.w3.org/DesignIssues/Notation3 and https://www.w3.org/TeamSubmission/n3/ into it.

**Comment by william-vw** [comment]
@RubenVerborgh Didn't you use a compiler-compiler to generate the parser? ..

**Comment by RubenVerborgh** [comment]
> Didn't you use a compiler-compiler to generate the parser?

I did for [a SPARQL parser](https://github.com/RubenVerborgh/SPARQL.js), but not for this library, given 1) performance and 2) the fact that it supports streaming parsing (for which it is hard to find parser generators). We want to parse files that are larger than main memory.

That said, this might create interesting requirements for Notation3, such that the meaning of a document cannot be changed by a `forAll` or so at the end of the document (I have no idea what the spec says now, if anything).

**Comment by jeswr** [comment]
All three items in this checklist are now implemented and tested. Formulas inside lists were added in #194 and literals as subjects in #193. N3 path syntax (the remaining unchecked item) is implemented in src/N3Parser.js (forward `!` and backward `^` paths, including mixed paths, in subject and object position) and covered by tests in test/N3Parser-test.js under the N3-format parser suite. Closing as resolved.
